src/dosbox.cpp

```diff
@@ -1125,9 +1125,11 @@ void DOSBOX_Init()
 	               "formats. If set to 0, the country code corresponding to the selected keyboard\n"
 	               "layout will be used.");
 
-	Pbool = secprop->Add_bool("expand_shell_variable", when_idle, false);
-	Pbool->Set_help("Enable expanding environment variables such as %PATH% in the DOS command shell\n"
-	                "(disabled by default).\n"
+	Pstring = secprop->Add_string("expand_shell_variable", when_idle, "auto");
+	const char *expand_shell_variable_choices[] = {"auto", "true", "false"};
```

Comment:
Close the setup list by adding a `, 0};` as the last element (another nuance of the setup class taking in literal arrays of  pointers; another thing for future refactoring).